metabase/metabase
# Improved encryption management

### Description

Improves the management of database encryption by:
- When initially enabling encryption on an existing database, all existing values that should be encrypted are immediately encrypted
- If a needed MB_ENCRYPTION_SECRET_KEY env variable is not set, halts server startup with a better error message
- If the wrong MB_ENCRYPTION_SECRET_KEY env variable is set, halts server start up with a better error message
- Adds an explicit "remove-encryption" CLI command rather than relying on the less-obvious `rotate-encryption-key ""` (which still works)

It also adds a new `remove-encryption` CLI command as a more obvious alternative to `rotate-encryption-key` passing an empty new key.

### How to verify

Check stored data in the following states:
- New server with no MB_ENCRYPTION_SECRET_KEY set **(data unencrypted)**
- Add MB_ENCRYPTION_SECRET_KEY to existing unencrypted database **(data encrypted)**
- Change MB_ENCRYPTION_SECRET_KEY value for existing encrypted server **(error)**
- Run existing encrypted server without MB_ENCRYPTION_SECRET_KEY **(error)**
- New server with MB_ENCRYPTION_SECRET_KEY set **(data encrypted)**
- Call `java -jar metabase.jar remove-encryption` **(data unencrypted)**

### Checklist
